cg/meta/upload/nf_analysis.py

```diff
@@ -46,7 +45,12 @@ def upload(self, ctx: click.Context, case: Case, restart: bool) -> None:
 
         # Scout specific upload
         if DataDelivery.SCOUT in case.data_delivery:
-            ctx.invoke(upload_to_scout, case_id=case.internal_id, re_upload=restart)
+            if case.data_analysis == Workflow.TOMTE:
+                ctx.invoke(
+                    upload_rna_to_scout, case_id=case.internal_id, analysis=case.data_analysis
+                )
```

Comment:
My guess your intention here is to invoke the upload_tomte command?

Reply:
However, I will still argue that this split in logic between Tomte and all other next-flow analyses warrants a child class. This change I want performed before a merge

Reply:
That is the one that I forgot to remove, see https://github.com/Clinical-Genomics/cg/pull/3678#pullrequestreview-2315592378

Reply:
There is no more split logic, but a direct cli call via `cg upload tomte-to-scout`

Reply:
Ok inheritance model implemented